### ISSUE TITLE
libpulseaudio-8.0-dev is added to LD_LIBRARY_PATH of Firefox

### ISSUE THREAD

**Issue by michalrus** [NixOS/nixpkgs#15126]
## Issue description

After setting `nixpkgs.config.pulseaudio = true`, Firefox should pick it up, as can be seen in https://github.com/NixOS/nixpkgs/blob/f6d7cefa7ba6429ec2a324ac3ae1fcefb45bdf0b/pkgs/applications/networking/browsers/firefox/wrapper.nix#L52.

However, `/nix/store/*-libpulseaudio-8.0-dev/lib` is added to the `LD_LIBRARY_PATH` in the wrapper script, instead of the compiled `/nix/store/*-libpulseaudio-8.0/lib`. Because of that, Firefox can’t see `libpulse*.so`’s and falls back to the ALSA sink.

Relevant change in my nixos-config is in https://github.com/michalrus/dotfiles/commit/875cbbe13d59bf2f82f3755967e40712dcdf7f83.

### Steps to reproduce

```
% cat $(which firefox) | grep pulse

export LD_LIBRARY_PATH=$LD_LIBRARY_PATH${LD_LIBRARY_PATH:+:}/nix/store/dmbwn5933n5q4pk3i4y9zb4vbxynf9aj-libpulseaudio-8.0-dev/lib
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH${LD_LIBRARY_PATH:+:}/nix/store/dmbwn5933n5q4pk3i4y9zb4vbxynf9aj-libpulseaudio-8.0-dev/lib64
```

## Technical details

* System: 16.09pre82105.7465bcd (Flounder)
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: "16.09pre82105.7465bcd"

**Comment by michalrus** [comment]
A quick workaround might be using:

```nix
# temporary solution for https://github.com/NixOS/nixpkgs/issues/15126
((wrapFirefox.override { libpulseaudio = libpulseaudio.out; }) firefox-unwrapped { })
```

… instead of `firefoxWrapper` directly.

**Comment by michalrus** [comment]
From #nixos @ Freenode:

> 00:44:02 < @michalrus > @joachifm: also, I couldn’t have made `wine` to work with `libpulseaudio`… It crashed with “cannot find `libpulse-common-8.0.so`: File not found” error. Might this be related?
> 00:44:21 < @michalrus > I’ve found mentions of `libpulseaudio-8.0-dev again`, in the `wine` binary.
> 00:44:35 < @michalrus> But it’s a statically linked binary… So dunno.

**Comment by joachifm** [comment]
I believe the wine issue is addressed by 784a5ed2ef4847d8b9a283c9a39bd2273bd709a8
